### PR TITLE
Deprecate env

### PR DESCRIPTION
**Pull request by jfindlay**
Replaces #31057.

### CODE REVIEW

**Comment by cachedout** [comment]
Before we merge this, I'd like to document it fully in the carbon release notes so that we don't forget.

**Comment by cachedout** [comment]
In these cases where we are already accepting kwargs anyway, I would like to be a little nicer and throw an warning stating that we're no longer accepting `env` and that the option will be ignored. That should at least help people debug mysterious failures.

File: salt/fileserver/__init__.py [hunk 1]
```diff
@@ -533,15 +523,6 @@ def serve_file(self, load):
...
         ret = {'data': '',
                'dest': ''}
-        if 'env' in load:
-            salt.utils.warn_until(
-                'Boron',
-                'Passing a salt environment should be done using \'saltenv\' '
-                'not \'env\'. This functionality will be removed in Salt '
-                'Boron.'
-            )
-            load['saltenv'] = load.pop('env')
```
Comment (cachedout) [inline salt/fileserver/__init__.py]: For all of these, we probably still want to check if `env` is in the load and delete it if so (while warning).

**Comment by jfindlay** [comment]
@cachedout, I have made all of the suggested changes.

**Pull request merged**